models/git_notes_publisher.rb

```diff
@@ -32,42 +32,38 @@ def prebuild(build, listener)
   # @param [Jenkins::Launcher] launcher the launcher that can run code on the node running this build
   # @param [Jenkins::Model::Listener] listener the listener for this build.
   def perform(build, launcher, listener)
-    BuildContext.instance.set(build, launcher, listener) do
-      notes = BuildNotes.new.notes
-      update_git_notes(notes)
-      notify_sqs(notes)
-    end
+    context = BuildContext.new(build, launcher, listener)
+    notes = BuildNotes.new(context).notes
+    update_git_notes(notes, context)
+    notify_sqs(notes, context)
   end
 
   private
 
-  def update_git_notes(notes)
-    git_updater = GitUpdater.new
+  def update_git_notes(notes, context)
+    git_updater = GitUpdater.new(context)
     retry_times = Constants::CONCURRENT_UPDATE_SLEEP_TIMES
     retry_times.each_with_index do |retry_time, idx|
       begin
-        info "updating git notes"
+        context.listener.info "updating git notes"
```

Comment:
Sorry, removing the mixin (since global state stored in the publisher is what got us stock here in the first place).